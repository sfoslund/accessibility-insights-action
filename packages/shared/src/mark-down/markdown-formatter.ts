// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { brand, brandLogoImg, toolName } from '../content/strings';
import { LogFormatter } from '../utils/LogFormatter';

export class MarkdownFormatter implements LogFormatter {
    public escaped = (markdown: string): string => {
        return markdown.replace(/</g, '\\<');
    };
    
    public snippet = (text: string): string => {
        return `\`${text}\``;
    };

    public static link = (href: string, text: string): string => {
        return `[${text}](${href})`;
    };

    public link: (href: string, text: string) => string = MarkdownFormatter.link;
    
    public static image = (altText: string, src: string): string => {
        return `![${altText}](${src})`;
    };
    
    public listItem = (text: string): string => {
        return `* ${text}`;
    };
    
    public heading = (text: string, headingLevel: number): string => {
        return `${'#'.repeat(headingLevel)} ${text}`;
    };
    
    public bold = (text: string): string => {
        return `**${text}**`;
    };
    
    public static productTitle = (): string => {
        return `${this.image(`${brand}`, brandLogoImg)} ${toolName}`;
    };

    public productTitle: () => string = MarkdownFormatter.productTitle;
    
    public footerSeparator = (): string => `---`;
    
    public sectionSeparator = (): string => '\n';
}
