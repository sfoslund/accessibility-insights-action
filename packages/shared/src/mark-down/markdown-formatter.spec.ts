// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { MarkdownFormatter } from "./markdown-formatter";


describe('MarkdownFormatter', () => {
    let testSubject: MarkdownFormatter;

    beforeAll(() => {
        testSubject = new MarkdownFormatter();
    })

    it('escaped', () => {
        expect(testSubject.escaped('<img>')).toEqual('\\<img>');
    });

    it('snippet', () => {
        expect(testSubject.snippet('code')).toMatchSnapshot();
    });

    it('link', () => {
        expect(testSubject.link('href', 'text')).toMatchSnapshot();
    });

    it('listItem', () => {
        expect(testSubject.listItem('li')).toMatchSnapshot();
    });

    it('heading', () => {
        expect(testSubject.heading('heading', 2)).toMatchSnapshot();
    });

    it('bold', () => {
        expect(testSubject.bold('text')).toMatchSnapshot();
    });

    it('productTitle', () => {
        expect(testSubject.productTitle()).toMatchSnapshot();
    });

    it('footerSeparator', () => {
        expect(testSubject.footerSeparator()).toMatchSnapshot();
    });

    it('sectionSeparator', () => {
        expect(testSubject.sectionSeparator()).toMatchSnapshot();
    });

    it('static link', () => {
        expect(MarkdownFormatter.link('href', 'text')).toMatchSnapshot();
    });

    it('static image', () => {
        expect(MarkdownFormatter.image('alt', 'src')).toMatchSnapshot();
    });

    it('static productTitle', () => {
        expect(MarkdownFormatter.productTitle()).toMatchSnapshot();
    });
});
