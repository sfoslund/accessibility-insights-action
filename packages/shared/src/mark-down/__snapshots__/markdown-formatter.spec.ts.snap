// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownFormatter bold 1`] = `"**text**"`;

exports[`MarkdownFormatter footerSeparator 1`] = `"---"`;

exports[`MarkdownFormatter heading 1`] = `"## heading"`;

exports[`MarkdownFormatter link 1`] = `"[text](href)"`;

exports[`MarkdownFormatter listItem 1`] = `"* li"`;

exports[`MarkdownFormatter productTitle 1`] = `"![Accessibility Insights](https://accessibilityinsights.io/img/a11yinsights-blue.svg) Accessibility Insights"`;

exports[`MarkdownFormatter sectionSeparator 1`] = `
"
"
`;

exports[`MarkdownFormatter snippet 1`] = `"\`code\`"`;

exports[`MarkdownFormatter static image 1`] = `"![alt](src)"`;

exports[`MarkdownFormatter static link 1`] = `"[text](href)"`;

exports[`MarkdownFormatter static productTitle 1`] = `"![Accessibility Insights](https://accessibilityinsights.io/img/a11yinsights-blue.svg) Accessibility Insights"`;
